core/templates/components/question-directives/questions-list/questions-list.component.html

```
@@ -36,17 +36,17 @@ <h4 *ngIf="difficultyCount != 1">Select Difficulty</h4>
-          <span (click)="removeSkill(skill.getId())" aria-hidden="true">
+          <span (click)="removeSkill(skill.getId())" aria-hidden="true" class="e2e-test-remove-skill-linkage-button">
             <i class="material-icons oppia-cross-icon md-18">&#xE14C;</i>
           </span>
-          <a [attr.href]="getSkillEditorUrl(skill.getId())" target="_blank" rel="noopener">
+          <a [attr.href]="getSkillEditorUrl(skill.getId())" target="_blank" rel="noopener" class="e2e-test-skill-linkage-description">
             {{skill.getDescription()}}
           </a>
         </div>
       </mat-card>
-      <button class="btn btn-success btn-sm" (click)="addSkill()" *ngIf="associatedSkillSummaries.length < MAX_SKILLS_PER_QUESTION">Link Another Skill</button>
+      <button class="btn btn-success btn-sm e2e-test-link-another-skill-button" (click)="addSkill()" *ngIf="associatedSkillSummaries.length < MAX_SKILLS_PER_QUESTION">Link Another Skill</button>
```

HTMLHint (1.9.2)

[warning] The type attribute must be present on <button> elements.

(button-type-require)